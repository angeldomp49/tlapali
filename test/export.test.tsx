import {Code} from '../index';
import renderer from 'react-test-renderer';
import { Lang, Theme } from '@makechtec/tlapali_dev';
import React from 'react';

test('testCode', () => {

  const exampleTheme: Theme = {
      name: 'exampleTheme',
      supportedLaguages: [
          {
              name: 'exampleLang',
              styleSets: [
                  {
                      rule: 'numbers',
                      styles: [
                          'color: red;'
                      ]
                  }
              ]
          }
      ]
  };

  const exampleLang: Lang = {
      name: 'exampleLang',
      keywords: [
          'public'
      ],
      rules: [
          {
              name: 'numbers',
              pattern: 'class\\s*(?<target>[A-z]*)\\s*\\{'
          }
      ]
  };

  const component = renderer.create(
      <Code theme={exampleTheme} lang={exampleLang}>
          {`
              public class App {
                  private static String name;
              }
          `}
      </Code>
  );

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});