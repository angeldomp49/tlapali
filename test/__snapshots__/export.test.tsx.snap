// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testCode 1`] = `
<div
  dangerouslySetInnerHTML={
    {
      "__html": "
              public class <span style="color: red;" >App</span> {
                  private static String name;
              }
          ",
    }
  }
/>
`;
